modules/issue_tracker/ajax/EditIssue.php

```diff
@@ -660,4 +658,64 @@ function getIssueFields()
               ];
 
     return $result;
+}
+
+/**
+ * If issueID is passed retrieves issue data from database,
+ * othewise return empty issue data object
+ *
+ * @param string $issueID the ID of the requested issue
+ *
+ * @return array
+ */
+function getIssueData($issueID)
+{
+
+    $user =& User::singleton();
+    $db   =& Database::singleton();
+
+    if ($issueID) {
+        return $db->pselectRow(
+            "SELECT i.*, c.PSCID, s.Visit_label as visitLabel FROM issues as i " .
+            "LEFT JOIN candidate c ON (i.candID=c.CandID)" .
+            "LEFT JOIN session s ON (i.sessionID=s.ID) " .
+            "WHERE issueID=:issueID",
+            ['issueID' => $issueID]
+        );
+    }
+
+    return [
+            'reporter'      => $user->getData('UserID'),
+            'dateCreated'   => date('Y-m-d H:i:s'),
+            'centerID'      => $user->getData('CenterID'),
+            'status'        => "new",
+            'priority'      => "normal",
+            'issueID'       => 0, //TODO: this is dumb
+            'title'         => null,
+            'lastUpdate'    => null,
+            'PSCID'         => null,
+            'assignee'      => null,
+            'history'       => null,
+            'watching'      => "Yes",
+            'visitLabel'    => null,
+            'category'      => null,
+            'lastUpdatedBy' => null,
+           ];
+}
+
+/**
+ * Utility function to return errors from the server
+ *
+ * @param string $message error message to display
+ *
+ * @return void
+ */
+function showError($message)
+{
+    if (!isset($message)) {
+        $message = 'An unknown error occurred!';
+    }
+    header('HTTP/1.1 500 Internal Server Error');
+    header('Content-Type: application/json; charset=UTF-8');
+    die(json_encode(['message' => $message]));
```

Comment:
This `die` will make it non-testable (calling die from inside of a test causing phpunit to exit), and it's probably best to not mix an exit side effect into a function with the name of `show`.

Can't the caller just return after calling this function? (or perhaps throw an exception if it's a truly exceptional situation..)